### PR TITLE
Create new route with faworite list

### PR DESCRIPTION
**Pull request by asavienko**

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -25,20 +27,61 @@ const StyledContent = styled(Content)`
...
 
 function App() {
+  const [rawList, setRawList] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const fetchCvList = async () => {
+    setLoading(true);
+    try {
+      const response = await fetch("/cvlist");
+      const obj = await response.json();
+      console.log(response);
```
Comment (bohdanSnappy) [inline src/App.js]: remove console.log()

File: src/App.js [hunk 2]
```diff
@@ -25,20 +27,61 @@ const StyledContent = styled(Content)`
...
+      console.log(obj);
+      if (obj.confirmation === "fail") {
+        throw new Error();
+      }
+      obj.data.forEach(cv => (cv.key = cv._id));
```
Comment (bohdanSnappy) [inline src/App.js]: use for of instead of forEach

File: src/App.js [hunk 3]
```diff
@@ -25,20 +27,61 @@ const StyledContent = styled(Content)`
...
+      openNotification({
+        type: "error",
+        message: "Не удалось загрузить данные"
+      });
+    }
+  };
+  const [favorites, setFavorites] = useState([]);
   return (
     <Router>
       <Layout>
```
Comment (bohdanSnappy) [inline src/App.js]: remove Layout

File: src/App.js [hunk 4]
```diff
@@ -25,20 +27,61 @@ const StyledContent = styled(Content)`
...
         </StyledHeader>
         <StyledDiv>
           <StyledContent>
             <Switch>
               <Route exact path="/" component={Home} />
-              <Route path="/favorites" />
-              <Route path="/list" component={CvTable} />
+              <Route
+                path="/favorites"
+                component={() => (
```
Comment (bohdanSnappy) [inline src/App.js]: check why do you need ()=>

File: src/actions/cvActions.js [hunk 5]
```diff
@@ -0,0 +1,5 @@
+import { SET_FAVORITE } from "../constants/constantsActions";
+
+export function setFavoriteList(favoriteList) {
```
Comment (bohdanSnappy) [inline src/actions/cvActions.js]:
```suggestion
export function setFavoriteList(favoriteCvList) {
```

File: src/actions/cvActions.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
...
+  return { type: SET_FAVORITE, favoriteCvList: favoriteList };
```
Comment (bohdanSnappy) [inline src/actions/cvActions.js]:
```suggestion
  return { type: SET_FAVORITE, favoriteCvList};
```

File: src/components/CvTable/CvList.js [hunk 7]
```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useState } from "react";
+import * as numeral from "numeral";
```
Comment (bohdanSnappy) [inline src/components/CvTable/CvList.js]: remove in future

File: src/components/CvTable/CvList.js [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+import { CvTable } from "../ReusableComponents/CvTable";
+import { connect } from "react-redux";
+import { setFavoriteList } from "../../actions/cvActions";
+import { EditFavoriteListButton } from "../ReusableComponents/Buttons";
+
+function CvList({
+  rawList,
+  fetchCvList,
+  loading,
+  setFavorites,
```
Comment (bohdanSnappy) [inline src/components/CvTable/CvList.js]: remove if unneeded

File: src/components/CvTable/CvList.js [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+  const [cvInfo, setCvInfo] = useState({ visible: false, cvInformation: {} });
+
+  useEffect(() => {
+    (async function() {
+      let cvList = rawList.length > 0 && rawList;
+      if (!cvList) {
+        cvList = await fetchCvList();
+      }
+      setCvList(cvList);
+      if (rawList && rawList.length) {
```
Comment (bohdanSnappy) [inline src/components/CvTable/CvList.js]: remove condition

File: src/components/CvTable/CvList.js [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+        const salary = rawList.map(({ salary }) => numeral(salary).value());
```
Comment (bohdanSnappy) [inline src/components/CvTable/CvList.js]: cvList is equal rawList

File: src/components/Favorites/Favorites.js [hunk 11]
```diff
@@ -0,0 +1,129 @@
...
+  const [addToFavoriteActive, setAddToFavoriteActive] = useState(false);
+  const [selectedRowKeys, setSelectedRowKeys] = useState([]);
+  const [salaryRange, setSalaryRange] = useState([]);
+  const [salaryFilterRange, setSalaryFilterRange] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [cvInfo, setCvInfo] = useState({ visible: false, cvInformation: {} });
+  const [cvList, setCvList] = useState([]);
+  const [filteredInfo, setFilteredInfo] = useState({});
+  useEffect(() => {
+    if (favoriteCvList && favoriteCvList.length) {
```
Comment (bohdanSnappy) [inline src/components/Favorites/Favorites.js]:
```suggestion
    if (favoriteCvList.length) {
```

File: src/components/Favorites/Favorites.js [hunk 12]
```diff
@@ -0,0 +1,129 @@
...
+        onSalaryRangeReset={onSalaryRangeReset}
+        onSalaryRangeChange={onSalaryRangeChange}
+        salaryRange={salaryRange}
+        loading={loading}
+        onRow={onRow}
+        handleChange={handleChange}
+        rowSelection={rowSelection}
+        filteredInfo={filteredInfo}
+      />
+      ;
```
Comment (bohdanSnappy) [inline src/components/Favorites/Favorites.js]: remove `;`

File: src/components/ReusableComponents/Buttons.js [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+
+export function EditFavoriteListButton({
+  addToFavoriteDisabled,
+  onPrimaryClick,
+  addToFavoriteActive,
+  cvCounts,
+  onCancelClick
+}) {
+  return (
+    <div style={{ height: 40 }}>
```
Comment (bohdanSnappy) [inline src/components/ReusableComponents/Buttons.js]: move style to styled components

File: src/components/ReusableComponents/CvTable.js [hunk 14]
```diff
@@ -0,0 +1,129 @@
...
+      key: "displayName"
+    },
+    {
+      title: "email",
+      dataIndex: "email",
+      key: "email",
+      filters: [{ text: "есть", value: true }, { text: "нет", value: false }],
+      filteredValue: filteredInfo.email || null,
+      onFilter: (value, { email }) =>
+        !!email ===
```
Comment (bohdanSnappy) [inline src/components/ReusableComponents/CvTable.js]:
```suggestion
        !!email === value
```

File: src/reducers/index.js [hunk 15]
```diff
@@ -0,0 +1,6 @@
+import {combineReducers} from "redux";
+import cvReducer from "./cvReducer";
+
```
Comment (bohdanSnappy) [inline src/reducers/index.js]: remove empty space

File: src/App.js (hunk 3 above)
Comment (asavienko) [reply src/App.js]: is it need to remove Header and Content elements from Layout?

File: src/App.js (hunk 4 above)
Comment (asavienko) [reply src/App.js]: i need it for passing props from parent component
i have changed Favorite, but in CvList props are nesesery

File: src/components/CvTable/CvList.js (hunk 9 above)
Comment (asavienko) [reply src/components/CvTable/CvList.js]: i changed rawList for cvList and i can`t remove condition

File: src/App.js [hunk 16]
```diff
@@ -23,26 +25,60 @@ const StyledContent = styled(Content)`
...
+    }
+  };
   return (
     <Router>
-      <Layout>
-        <StyledHeader
-          theme="light"
-        >
+      <StyledLayout>
+        <StyledHeader theme="light">
```
Comment (bohdanSnappy) [inline src/App.js]: remove Header. left <div> or <header>

File: src/components/ReusableComponents/Buttons.js [hunk 17]
```diff
@@ -0,0 +1,41 @@
+import { Button } from "antd";
+import React from "react";
+import styled from "styled-components";
+
+const StyledCloseButton = styled(Button)`
+  margin-left: 5px;
+`;
+
+const StyledDivWrapper = styled.div`
+height: 40px;
```
Comment (bohdanSnappy) [inline src/components/ReusableComponents/Buttons.js]: add tab

File: src/components/ReusableComponents/CvTable.js [hunk 18]
```diff
@@ -0,0 +1,126 @@
...
+      title: "Имя",
+      dataIndex: "displayName",
+      key: "displayName"
+    },
+    {
+      title: "email",
+      dataIndex: "email",
+      key: "email",
+      filters: [{ text: "есть", value: true }, { text: "нет", value: false }],
+      filteredValue: filteredInfo.email || null,
```
Comment (bohdanSnappy) [inline src/components/ReusableComponents/CvTable.js]: check null or "

File: src/components/CvTable/CvList.js (hunk 9 above)
Comment (asavienko) [reply src/components/CvTable/CvList.js]: this part of code was change in other branch(work_with_favorites) according to comment

**Pull request merged**